Source/hydro/Castro_ctu_nd.F90

```diff
@@ -104,245 +92,95 @@ subroutine umeth(q, qd_lo, qd_hi, &
-
-    ! Local constants
-    dxinv = ONE/dx(1)
-    dtdx = dt*dxinv
-    hdtdx = HALF*dtdx
-
-#if AMREX_SPACEDIM >= 2
-    dyinv = ONE/dx(2)
-    dtdy = dt*dyinv
-    hdtdy = HALF*dtdy
-#endif
-
-#if AMREX_SPACEDIM == 3
-    dzinv = ONE/dx(3)
-    dtdz = dt*dzinv
-    hdtdz = HALF*dtdz
-    cdtdx = dtdx*THIRD
-    cdtdy = dtdy*THIRD
-    cdtdz = dtdz*THIRD
-#endif
+    logical :: compute_shock
 
     hdt = HALF*dt
 
-    fglo = lo - dg(:)  ! face + one ghost
-    fghi = hi + 2*dg(:)
-
-    glo = lo - dg(:)  ! one ghost,  this can be used for face-based arrays too
-    ghi = hi + dg(:)
-
-    call bl_allocate ( qxm, fglo, fghi, NQ)
-    call bl_allocate ( qxp, fglo, fghi, NQ)
-
-#if AMREX_SPACEDIM >= 2
-    call bl_allocate ( qym, fglo, fghi, NQ)
-    call bl_allocate ( qyp, fglo, fghi, NQ)
-#endif
-
-#if AMREX_SPACEDIM == 3
-    call bl_allocate ( qzm, fglo, fghi, NQ)
-    call bl_allocate ( qzp, fglo, fghi, NQ)
-#endif
-
-    if (ppm_type .gt. 0) then
-       ! x-index, y-index, z-index, dim, characteristics, variables
-       call bl_allocate ( Ip, glo(1),ghi(1),glo(2),ghi(2),glo(3),ghi(3),1,AMREX_SPACEDIM,1,3,1,NQ)
-       call bl_allocate ( Im, glo(1),ghi(1),glo(2),ghi(2),glo(3),ghi(3),1,AMREX_SPACEDIM,1,3,1,NQ)
-
-       ! for source terms
-       call bl_allocate ( Ip_src, glo(1),ghi(1),glo(2),ghi(2),glo(3),ghi(3),1,AMREX_SPACEDIM,1,3,1,QVAR)
-       call bl_allocate ( Im_src, glo(1),ghi(1),glo(2),ghi(2),glo(3),ghi(3),1,AMREX_SPACEDIM,1,3,1,QVAR)
-
-       ! for gamc -- needed for the reference state in eigenvectors
-       call bl_allocate ( Ip_gc, glo(1),ghi(1),glo(2),ghi(2),glo(3),ghi(3),1,AMREX_SPACEDIM,1,3,1,1)
-       call bl_allocate ( Im_gc, glo(1),ghi(1),glo(2),ghi(2),glo(3),ghi(3),1,AMREX_SPACEDIM,1,3,1,1)
-    else
-       call bl_allocate ( dqx, glo, ghi, NQ)
-#if AMREX_SPACEDIM >= 2
-       call bl_allocate ( dqy, glo, ghi, NQ)
-#endif
-#if AMREX_SPACEDIM == 3
-       call bl_allocate ( dqz, glo, ghi, NQ)
-#endif
-    end if
-
-    ! for the hybrid Riemann solver
-    call bl_allocate(shk, glo, ghi)
-
     ! multidimensional shock detection
 
 #ifdef SHOCK_VAR
-    uout(lo(1):hi(1),lo(2):hi(2),lo(3):hi(3),USHK) = ZERO
-
-    call ca_shock(lo-dg, hi+dg, &
-                  q, qd_lo, qd_hi, &
-                  shk, glo, ghi, &
-                  dx)
-
-    ! Store the shock data for future use in the burning step.
-
-    do k = lo(3), hi(3)
-       do j = lo(2), hi(2)
-          do i = lo(1), hi(1)
-             uout(i,j,k,USHK) = shk(i,j,k)
-          enddo
-       enddo
-    enddo
-
-    ! Discard it locally if we don't need it in the hydro update.
-
-    if (hybrid_riemann /= 1) then
-       shk(:,:,:) = ZERO
-    endif
+    compute_shock = .true.
 #else
+    compute_shock = .false.
+#endif
+
     ! multidimensional shock detection -- this will be used to do the
     ! hybrid Riemann solver
-    if (hybrid_riemann == 1) then
-       call ca_shock(lo-dg, hi+dg, &
+    if (hybrid_riemann == 1 .or. compute_shock) then
```

Comment:
Previously, we would zero out the `shk` data if `SHOCK_VAR` is defined but `hybrid_riemann /= 1`. I think this changes the logic, because now `shk` will have non-zero data if `SHOCK_VAR` is defined regardless of the value of `hybrid_riemann`. Not sure if that has any impact down the line.